website/blog/2021-03-27-traffic-split-in-apache-apisix-ingress-controller.md

```diff
@@ -0,0 +1,177 @@
+
++ nginx.ingress.kubernetes.io/canary-by-header-pattern
+
+Similar to nginx.ingress.kubernetes.io/canary-by-header, but the value is a [PCRE](https://www.pcre.org/) compatible regular expression.
+
++ nginx.ingress.kubernetes.io/canary-by-cookie
+
+Use data field in Cookie header to decide the backend service.
+
++ nginx.ingress.kubernetes.io/canary-weight
+
+Assign weight value between 0 and 100, traffic will be delivered according to this weight, a 0 weight means all traffic will be routed to the canary app and 100 weight will route all traffic to the stable app.
+
+The following YAML snippet proxies requests whose URI path is led by "/get" and the User-Agent matches with the ".*Mozilla.*" pattern to the canary app "foo-canary".
+
+```
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+      kubernetes.io/ingress.class: nginx
+      nginx.ingress.kubernetes.io/canary: "true"
+      nginx.ingress.kubernetes.io/canary-by-header: "User-Agent"
+      nginx.ingress.kubernetes.io/canary-by-header-pattern:
+".*Mozilla.*"
+  name: ingress-v1beta1
+```
+
+## Kong
+
+The Kong Gateway has a [canary release plugin](https://docs.konghq.com/hub/kong-inc/canary/0.32-x.html) and exposes this plugin to its ingress controller by [KongPlugin](https://docs.konghq.com/hub/) resource. Administrators/Users need to create a KongPlugin object and fill the canary release rule, injecting an annotation "konghq.com/plugins" to the target [Kuberentes Service](https://kubernetes.io/docs/concepts/services-networking/service/). Or you can create a [KongClusterPlugin](https://docs.konghq.com/kubernetes-ingress-controller/1.1.x/guides/using-kongclusterplugin-resource/) object to let this canary rule effective in the whole cluster.
+
+```
+apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  name: foo-canary
+config:
+  percentage: 30
+  upstream_host: foo.com
+  upstream_fallback: false
+  upstream_port: 80
+plugin: canary
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: foo-canary
+  labels:
+    app: foo
+  annotations:
+    konghq.com/plugins: foo-canary
+spec:
+  ports:
+  - port: 80
+    targetPort: 80
+    protocol: TCP
+    name: http
+  selector:
+      app: foo
+      canary: true
+```
+
+The above case marks the service "foo-canary" as "canary", and creates a canary release rule to proxy 30% traffic to this service.
+
+## Apache APISIX
+
+[Apache APISIX](https://apisix.apache.org) splits traffic with custom rules by its [traffic-split](apisix.apache.org/docs/apisix/plugins/traffic-split) plugin, and Apache APISIX Ingress Controller implements the traffic split feature to [ApisixRoute](apisix.apache.org/docs/ingress-controller/concepts/apisix_route) (as the first-class support, without relying on annotations) by this plugin and the flexible route match abilities in ApisixRoute.
```

Comment:
Hi @iamayushdas 

I think `traffic-split` and `ApisixRoute` links should add the **https://**

Reply:
Ya you are right